modules/standard/GMP.chpl

```diff
@@ -560,7 +560,7 @@ module GMP {
     }
     proc set_str(str:string, base:int=0)
     {
-      on this do mpz_set_str(this.mpz, str.c_str(), base.safeCast(c_int));
+      on this do mpz_set_str(this.mpz, str.localize().c_str(), base.safeCast(c_int));
```

Comment:
> Intuitively, I'm guessing it to make sure that the resulting c string is local to the locale in question.

Yup.

>  What happens if they aren't used?

You will be accessing a potentially garbage pointer, and will most likely segfault.

> What happened in master prior to this PR?

We copy the string and leak it.

There really isn't too much of an option on this one unfortunately. The only thing I can think of to remove the localize calls is to just insert them automatically in the compiler whenever we call `.c_str()`. But that isn't very appealing to me.